### PR TITLE
[CARBONDATA-708] Fixed Between Filter Issue for Pushed Filters in case it was applied …

### PR DESCRIPTION
**Pull request by bhavya411**
Fixed Between Filter Issue for Pushed Filters in case it was applied on Dimension
Test
Add Examples of Between Filter
Removed Scalastyle 
Manually Test BETWEEN Filter
Run Unit Test Cases

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Can one of the admins verify this patch?

**Comment by chenliang613** [comment]
add to whitelist

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/982/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/985/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/993/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1015/

**Comment by ravipesala** [comment]
Please rebase the code.

File: integration/spark2/src/main/scala/org/apache/carbondata/spark/CarbonFilters.scala [hunk 1]
```diff
@@ -110,8 +110,16 @@ object CarbonFilters {
     }
 
     def getCarbonLiteralExpression(name: String, value: Any): CarbonExpression = {
-      new CarbonLiteralExpression(value,
-        CarbonScalaUtil.convertSparkToCarbonDataType(dataTypeOf(name)))
+      val dataTypeOfAttribute = CarbonScalaUtil.convertSparkToCarbonDataType(dataTypeOf(name))
+      val dataType = if (Option(value).isDefined
+        && dataTypeOfAttribute == DataType.STRING
+        && value.isInstanceOf[Double]) {
```
Comment (ravipesala) [inline integration/spark2/src/main/scala/org/apache/carbondata/spark/CarbonFilters.scala]: Can you explain with out this fix what is the issue happening?

Comment (bhavya411) [inline integration/spark2/src/main/scala/org/apache/carbondata/spark/CarbonFilters.scala]: In the scenarios where we have stored the numbers as String (and have included it as a Dimension) and if we are trying to do operations like >= or <= or between it is not working correctly 
and it gives some random results as it is comparing a String with a Double. So for those cases this fix will work.

**Comment by bhavya411** [comment]
I have rebased the code

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1139/

File: examples/spark2/src/main/scala/org/apache/carbondata/examples/CarbonSelectFilterExample.scala [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+import java.io.File
+
+import org.apache.spark.sql.SparkSession
+
+import org.apache.carbondata.core.constants.CarbonCommonConstants
+import org.apache.carbondata.core.util.CarbonProperties
+
+
+
+object CarbonSelectFilterExample {
```
Comment (ravipesala) [inline examples/spark2/src/main/scala/org/apache/carbondata/examples/CarbonSelectFilterExample.scala]: Please move this to common-test

**Comment by ravipesala** [comment]
@bhavya411 thanks for working on this.  LGTM but please move the examples to testcase.

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1304/

**Comment by bhavya411** [comment]
@ravipesala Created a New PR #697 for the fixes that were suggested , added a Unit test class to make sure it runs everytime with the build and removed the example

**Pull request closed without merging**